Sources/Media/IOAudioResampler.swift

```diff
@@ -10,13 +10,68 @@ protocol IOAudioResamplerDelegate: AnyObject {
     func resampler(_ resampler: IOAudioResampler<Self>, errorOccurred error: AudioCodec.Error)
 }
 
+struct IOAudioResamplerSettings {
+    let sampleRate: Float64
+    let channels: UInt32
+    let downmix: Bool
+    let channelMap: [NSNumber]?
+
+    init(sampleRate: Float64 = 0, channels: UInt32 = 0, downmix: Bool = false, channelMap: [NSNumber]? = nil) {
+        self.sampleRate = sampleRate
+        self.channels = channels
+        self.downmix = downmix
+        self.channelMap = channelMap
+    }
+
+    func invalidate(_ oldValue: IOAudioResamplerSettings!) -> Bool {
+        return (sampleRate != oldValue.sampleRate &&
```

Comment:
Does this need to have OR condition? Please take look, thanks.

Reply:
You're right. https://github.com/shogo4405/HaishinKit.swift/commit/780449bc2b36f2acaa3df1fdaf56ac65bc547f72